content/tutorials/build-apps/build-a-browser-wallet-in-js.md

```diff
@@ -0,0 +1,194 @@
+
+7. Set up a Vite bundler. Create a file named `vite.config.js` in the root directory of the project and fill it with the following code:
+
+{{ include_code("_code-samples/build-a-wallet/js/vite.config.js", language="js") }}
+
+This example includes the necessary configuration to make [xrpl.js work with Vite](https://github.com/XRPLF/xrpl.js/blob/main/UNIQUE_SETUPS.md#using-xrpljs-with-vite-react).
+
+8. Add script to `package.json`
+
+In your `package.json` file, add the following section if it's not there already:
+
+```json
+"scripts": {
+    "dev": "vite"
+}
+```
+
+### 2. Create the Home Page (Displaying Account & Ledger Details)
+
+In this step, we create a home page that displays account and ledger details.
+
+![Home Page Screenshot](img/js-wallet-home.png)
+
+1. Make a new folder named `src` in the root directory of the project.
+
+2. If not already present, create new files in the root folder named `index.html`, `index.js` and `index.css`.
+
+3. Copy the contents of [index.html]({{target.github_forkurl}}/tree/{{target.github_branch}}/content/_code-samples/build-a-wallet/js/index.html) in your code.
+
+4. Add styling to your [index.css]({{target.github_forkurl}}/tree/{{target.github_branch}}/content/_code-samples/build-a-wallet/js/index.css) file by following the link.
+
+This basic setup creates a homepage and applies some visual styles. The goal is for the homepage to:
+
+- Display our account info
+- Show what's happening on the ledger
+- And add a little logo for fun
+
+To make that happen, we need to connect to the XRP Ledger and look up the account and the latest validated ledger.
+
+5. In the `src/` directory, make a new folder named `helpers`. Create a new file there named `get-wallet-details.js` and define a function named `getWalletDetails` there. This function uses the [account_info method](account_info.html) to fetch account details and the [server_info method](server_info.html) to calculate the current [reserves](reserves.html). The code to do all this is as follows:
+
+{{ include_code("_code-samples/build-a-wallet/js/src/helpers/get-wallet-details.js", language="js") }}
+
+6. Now, let's add the code to `index.js` file to fetch the account and ledger details and display them on the home page. Copy the code written below to the `index.js` file. Here we render the wallet details using the function we defined in `get-wallet-details.js`. In order to make sure we have up to date ledger data, we are using the [ledger stream](subscribe.html#ledger-stream) to listen for ledger close events.
+
+{{ include_code("_code-samples/build-a-wallet/js/index.js", language="js") }}
+
+The one other thing we do here is add events to two buttons - one to send XRP and one to view transaction history. They won't work just yet—we'll implement them in the next steps.
+
+Now the application is ready to run. You can start it in dev mode using the following command:
+
+```bash
+yarn dev
+```
```

Comment:
At this point the code crashes because `index.js` refers to `./src/helpers/render-xrpl-logo` but we haven't defined that yet.
<img width="1408" alt="image" src="https://user-images.githubusercontent.com/19694186/235007399-4155e403-e270-42e2-a2e9-bc011be09417.png">

Reply:
After adding BOTH `render-xrpl-logo.js` and `xrpl.svg` it renders properly.

Reply:
I've added this: 

Also, add the following two files, one in the `helpers` folder [render-xrpl-logo.js]({{target.github_forkurl}}/tree/{{target.github_branch}}/content/_code-samples/build-a-wallet/js/src/helpers/render-xrpl-logo.js) and for the other one create a new folder named `assets` in `src` directory and add [xrpl.svg]({{target.github_forkurl}}/tree/{{target.github_branch}}/content/_code-samples/build-a-wallet/js/src/assets/xrpl.svg) file. These files are used to render the XRPL logo for aesthetic purposes.

Reply:
Works up to here following steps as written now :)